src/test/java/com/es/phoneshop/controller/ProductListPageServletTest.java

```diff
@@ -25,17 +24,21 @@
     @Mock
     private RequestDispatcher requestDispatcher;
 
-    private ProductListPageServlet servlet = new ProductListPageServlet();
+    private final ProductListPageServlet servlet = new ProductListPageServlet();
 
     @Before
-    public void setup(){
+    public void setup() throws ServletException, IOException {
         when(request.getRequestDispatcher(anyString())).thenReturn(requestDispatcher);
+        servlet.doGet(request, response);
```

Comment:
The actual invocations of method(s) should be put to test methods, not setup one. Setup method is intended for mocks setup and when-then adjustments